apis/python/src/tiledbsoma/io/ingest.py

```diff
@@ -1722,6 +1788,7 @@ def add_matrix_to_collection(
     ingest_mode: IngestMode = "write",
     use_relative_uri: bool | None = None,
     context: SOMATileDBContext | None = None,
+    schema_validation: bool = False,
```

Comment:
Do we intend for this param to be `False` by default?  It seems like this parameter should be `True` by default, given the nature of the check.

Or if we are looking to migrate users over time, we could:
* Always: issue a warning if the schema fails to validate
* Flip this param to `True` at the next major release (3.0)

@aaronwolen - thoughts?

Reply:
I thought that since the stage of the function is `maturing` I did not want to go for throttle on that, risking breaking API backwards compatibility. Of course I could set it to `True` if we decide that we want to hard impose it.